src/qutip_qip/decompose/decompose_single_qubit_gate.py

```diff
@@ -0,0 +1,256 @@
+        targets=[0],
+        arg_value=theta,
+        arg_label=r"{:0.2f} \times \pi".format(theta / np.pi),
+    )
+    Rz_beta = Gate(
+        "RZ",
+        targets=[0],
+        arg_value=beta,
+        arg_label=r"{:0.2f} \times \pi".format(beta / np.pi),
+    )
+
+    return (Rz_alpha, Rx_theta, Rz_beta, Phase_gate)
+
+
+# Functions for ABC_decomposition
+
+
+def _ZYZ_pauli_X(input_gate):
+    """Returns a 1 qubit unitary as a product of ZYZ rotation matrices and
+    Pauli X."""
+    check_gate(input_gate, num_qubits=1)
+    alpha, theta, beta, global_phase_angle = _angles_for_ZYZ(input_gate)
+
+    Phase_gate = Gate(
+        "GLOBALPHASE",
+        targets=[0],
+        arg_value=global_phase_angle,
+        arg_label=r"{:0.2f} \times \pi".format(global_phase_angle / np.pi),
+    )
+    Rz_A = Gate(
+        "RZ",
+        targets=[0],
+        arg_value=alpha,
+        arg_label=r"{:0.2f} \times \pi".format(alpha / np.pi),
+    )
+    Ry_A = Gate(
+        "RY",
+        targets=[0],
+        arg_value=theta / 2,
+        arg_label=r"{:0.2f} \times \pi".format(theta / np.pi),
+    )
+    Pauli_X = Gate("X", targets=[0])
+    Ry_B = Gate(
+        "RY",
+        targets=[0],
+        arg_value=-theta / 2,
+        arg_label=r"{:0.2f} \times \pi".format(-theta / np.pi),
+    )
+    Rz_B = Gate(
+        "RZ",
+        targets=[0],
+        arg_value=-(alpha + beta) / 2,
+        arg_label=r"{:0.2f} \times \pi".format(-(alpha + beta) / (2*np.pi)),
+    )
+    Rz_C = Gate(
+        "RZ",
+        targets=[0],
+        arg_value=(-alpha + beta) / 2,
+        arg_label=r"{:0.2f} \times \pi".format((-alpha + beta) / (2*np.pi)),
+    )
+
+    return (Rz_A, Ry_A, Pauli_X, Ry_B, Rz_B, Pauli_X, Rz_C, Phase_gate)
+
+
+_single_decompositions_dictionary = {
+    "ZYZ": _ZYZ_rotation,
+    "ZXZ": _ZXZ_rotation,
+    "ZYZ_PauliX": _ZYZ_pauli_X,
+}  # other combinations to add here
+
+
+def decompose_one_qubit_gate(input_gate, method):
+    r""" An input 1-qubit gate is expressed as a product of rotation matrices
+    :math:`\textrm{R}_i` and :math:`\textrm{R}_j` or as a product of rotation
+    matrices :math:`\textrm{R}_i` and :math:`\textrm{R}_j` and a
+    Pauli :math:`\sigma_k`.
+
+    Here, :math:`i \neq j` and :math:`i, j, k \in {x, y, z}`.
+
+    Based on Lemma 4.1 and Lemma 4.3 of
+    https://arxiv.org/abs/quant-ph/9503016v1 respectively.
+
+    .. math::
+
+        U = \begin{bmatrix}
+            a       & b \\
+            -b^*       & a^*  \\
+            \end{bmatrix} = \textrm{R}_i(\alpha)  \textrm{R}_j(\theta) \textrm{R}_i(\beta) = \textrm{A} \sigma_k \textrm{B} \sigma_k \textrm{C}
+
+    Here,
+
+    * :math:`\textrm{A} = \textrm{R}_i(\alpha) \textrm{R}_j\left(\frac{\theta}{2} \right)`
+
+    * :math:`\textrm{B} = \textrm{R}_j \left(\frac{-\theta}{2} \right)\textrm{R}_i \left(\frac{- \left(\alpha + \beta \right)}{2} \right)`
+
+    * :math:`\textrm{C} = \textrm{R}_i \left(\frac{\left(-\alpha + \beta\right)}{2} \right)`
```

Comment:
Lines too long. Split in math mode.

Reply:
If you find the math looks better this way, we can also keep it, although it avoids PEP8. Personally, I think no one will read it in the source code. It is for the sphinx anyway. Maybe we can later move equations to the main docs if we want to have a proper doc for decomposition.

Reply:
It's up to you. I was trying to keep my code consistent with PEP8 but if you would prefer to ignore the doctsrings, it's ok with me. 

> Maybe we can later move equations to the main docs if we want to have a proper doc for decomposition.

This could work as well.

Reply:
Oh, in general, we should keep also docstrings PEP8 compatible. If you find a way to make the math compatible with PEP8 without destroying the readability then please surely do it. I was just saying that if there is no easy way then we can also keep it like this.

Reply:
> If you find a way to make the math compatible with PEP8 without destroying the readability then please surely do it.

I don't know an easy way yet.